sdk/core/azure_core/src/http/pager.rs

```diff
@@ -289,6 +297,7 @@ impl<P: Page> fmt::Debug for ItemIterator<P> {
 pub struct PageIterator<P> {
     #[pin]
     stream: Pin<BoxedStream<P>>,
+    next: Arc<Mutex<Option<String>>>,
```

Comment:
Feels like a place where [`AtomicCell`](https://docs.rs/crossbeam/latest/crossbeam/atomic/struct.AtomicCell.html) would be useful, but I wouldn't want the dependency on crossbeam just for that. Plus, this should be fine for now, and it's all internal so we can adjust later.